rtengine/ahd_demosaic_RT.cc

```diff
@@ -0,0 +1,229 @@
+                for (int row = top + 1; row < top + TS - 1 && row < height - 3; row++) {
+                    int cng = FC(row + 1, FC(row + 1, 0) & 1);
+                    for (int col = left + 1; col < std::min(left + TS - 1, width - 3); col++) {
+                        pix = &(rawData[row][col]);
+                        rix = &rgb[d][row - top][col - left];
+                        lix = &lab[d][row - top][col - left];
+                        if (FC(row, col) == 1) {
+                            rix[0][2 - cng] = CLIP(pix[0] + (0.5f * (pix[-1] + pix[1]
+                                                       - rix[-1][1] - rix[1][1] ) ));
+                            rix[0][cng] = CLIP(pix[0] + (0.5f * (pix[-width] + pix[width]
+                                                       - rix[-TS][1] - rix[TS][1])));
+                            rix[0][1] = pix[0];
+                        } else {
+                            rix[0][cng] = CLIP(rix[0][1] + (0.25f * (pix[-width - 1] + pix[-width + 1]
+                                                        + pix[+width - 1] + pix[+width + 1]
+                                                        - rix[-TS - 1][1] - rix[-TS + 1][1]
+                                                        - rix[+TS - 1][1] - rix[+TS + 1][1])));
+                            rix[0][2 - cng] = pix[0];
+                        }
+                        float xyz0 = 0.f;
+                        float xyz1 = 0.f;
+                        float xyz2 = 0.f;
+                        
+                        for(unsigned int c = 0; c < 3; ++c) {
+                            xyz0 += xyz_cam[0][c] * rix[0][c];
+                            xyz1 += xyz_cam[1][c] * rix[0][c];
+                            xyz2 += xyz_cam[2][c] * rix[0][c];
+                        }
+
+                        xyz0 = cbrt[xyz0];
+                        xyz1 = cbrt[xyz1];
+                        xyz2 = cbrt[xyz2];
+
+                        lix[0][0] = 116.f * xyz1 - 16.f;
+                        lix[0][1] = 500.f * (xyz0 - xyz1);
+                        lix[0][2] = 200.f * (xyz1 - xyz2);
+                    }
+                }
+
+            //  Build homogeneity maps from the CIELab images:
+
+            for (int row = top + 2; row < top + TS - 2 && row < height - 4; row++) {
+                int tr = row - top;
+
+                for (int col = left + 2, tc = 2; col < left + TS - 2 && col < width - 4; col++, tc++) {
+                    for (int d = 0; d < 2; d++) {
+                        lix = &lab[d][tr][tc];
+
+                        for (int i = 0; i < 4; i++) {
+                            ldiff[d][i] = std::fabs(lix[0][0] - lix[dir[i]][0]);
+                            abdiff[d][i] = SQR(lix[0][1] - lix[dir[i]][1])
+                                           + SQR(lix[0][2] - lix[dir[i]][2]);
+                        }
+                    }
+
+                    float leps = min(max(ldiff[0][0], ldiff[0][1]),
```

Comment:
Here you use `rtengine::(min|max)()`, while above it's `std::(min|max)()`. There's a unification due, IMHO.

Reply:
Fixed